h2o-core/src/main/java/water/DKV.java

```diff
@@ -185,7 +185,6 @@ static void write_barrier() {
   static public void prefetch( String key_name ) {        get(Key.make(key_name),false); }
 
   static private Value get( Key key, boolean blocking ) {
-    if (key == null) return null;
```

Comment:
the reasons are:
  - the method is private - the caller is DKV class itself, it has to handle null before calling this method 
 - performance impact - probably not big deal, but without proper perf test we can say nothing about  it
 - the commit was hidden in POJO commit - we need to be careful on changing core as a side effect of other changes, it is better to make explicit PR with core change